### DISCUSSION TITLE
Is there a shortcut to insert a dash?

### DISCUSSION THREAD

**Discussion by ralf3u** [geany/geany#3887] [Q&A]
When typing text in Geany, I use often a dash. Till now I use LibreOffice for this by creating a dash to copy and paste it in Geany. So, I asked myself if there is not a faster way.

In LibreOffice, if I type this:
`hello --world !`
the text will be automatically transformed to this:
`hello –world !`
So, the two hyphens will be transformed to a dash. This is the way how I produce a dash in LibreOffice.

Is there a shortcut to insert a dash in Geany?

In a famous search engine I searched for this: geany dash shortcut.
But I couldn't find something useful to solve the question.

Also there I couldn't find something useful to solve the question:
https://wiki.geany.org/howtos/sorted_keyboard_shortcuts

**Comment by elextr** [comment]
Presuming your "dash" is actually an "em-dash" (the things you put two of are technically "dash"es) then `ctrl+shift+u`2014`enter` will insert one.

**Comment by elextr** [comment]
Making such into a snippet with an appropriate keybinding is left as an exercise for the reader of the manual :grinning:

**Comment by b4n** [comment]
Another solution is to have a keyboard layout with this available, e.g. `fr:oss` does (e.g. <kbd>AltGr + Shift + 4</kbd> for the EM Dash and <kbd>AltGr + Shift + 5</kbd> for the EN Dash). There is also the [Compose](https://en.wikipedia.org/wiki/Compose_key) sequence <kbd>Compose - - -</kbd> for EM Dash.

**Reply by elextr** [reply]
Meanwhile, back in the world of US-105 keyboards ... :frowning_face:

**Comment by ralf3u** [comment]
@elextr
Thanks to your comment I found more characters here:
https://en.wikipedia.org/wiki/List_of_Unicode_characters

I just added on the following webpage one row — it is the second shortcut of the table title "Editing":
https://wiki.geany.org/howtos/sorted_keyboard_shortcuts

There I wrote the following text:
```
Ctrl-Shift-U, Unicode, Enter
Insert one special character
Hold down the Ctrl and Shift keys together, press the U key, release the U key, release the Ctrl and Shift keys, type for example 2014, and then hit the Enter key.
```
Thank you so much for your help.

**Reply by elextr** [reply]
Its actually in the [manual](https://www.geany.org/manual/current/index.html#inserting-unicode-characters), although the second paragraph about unspecified "earlier versions" of Geany is probably irrelevant now and could be removed.

**Comment by ralf3u** [comment]
@b4n
At the moment I am not using a French keyboard layout (I use a French keyboard layout when I visit my mother, who is French), but a German keyboard layout. Thanks to your comment I played a little bit around and found this solution for me:
`Alt Gr` + `-`
This is for me the best way to insert a dash.

Thank you so much for your help.

**Comment by ralf3u** [comment]
I just added two new shortcuts on the following webpage:
https://wiki.geany.org/howtos/sorted_keyboard_shortcuts

In the table with the title "Editing" I wrote this text to make the list complete:
```
|Alt Gr-character|Insert one special character|Press on the keyboard at the same time the Alt Gr key and one symbol key, like a number or a letter.|
|Alt Gr-Shift-character|Insert one special character|Press on the keyboard at the same time the Alt Gr key, the Shift key and one symbol key, like a number or a letter.|
```

**Reply by elextr** [reply]
No, those are not Geany shortcuts, they are system key sequences and are keyboard specific, as I implied [above](https://github.com/geany/geany/discussions/3887#discussioncomment-9564508) they are not available on common keyboards.  These are not Geany specific so they do not belong in a table of Geany shortcuts.

**Comment by ralf3u** [comment]
>No, those are not Geany shortcuts, they are system key sequences and are keyboard specific, as I implied https://github.com/geany/geany/discussions/3887#discussioncomment-9564508 they are not available on common keyboards. These are not Geany specific so they do not belong in a table of Geany shortcuts.

The title of the article at geany.org is "Keyboard shortcuts". In that article there is a section called "Built-in shortcuts". So, my idea is to add at the end of the article a new section called "Keyboard specific shortcuts". The text could be like this: "Some keyboards allow further shortcuts to insert one single character." Then I would add a table with the two shortcuts that are executed with the Alt Gr key. The shortcut "Ctrl-Shift-U, Unicode, Enter" will stay in the "Built-in shortcuts" section.

Is that OK?

**Reply by elextr** [reply]
Yeah, so long as its clear that this relates to the operating system and particular hardware keyboards thats fine.

Note, I am being totally selfish here, I don't want to be fielding complaints that some binding doesn't work on particular keyboards or systems ;-)

Maybe something like "Shortcuts in this section are _not_ part of Geany but provided by some systems and may be useful in Geany.  Some of these may depend on particular keyboard layouts and may not be available on others." as the first line of the new section.

**Comment by ralf3u** [comment]
Done.

https://wiki.geany.org/howtos/sorted_keyboard_shortcuts

If something is not correct, please leave me a message.

**Comment by ralf3u** [comment]
I just added this text:

So, it can be that those shortcuts don't work on some systems and on some keyboards.
